spec/core/data_structures.md

```diff
@@ -4,195 +4,239 @@ Here we describe the data structures in the Tendermint blockchain and the rules
+|------------|----------------------------------|----------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------|
+| Height     | int64                            | Height at which this commit was created.                              | Must be > 0                                                                                              |
+| Round      | int32                            | Round that the commit corresponds to.                                | Must be > 0                                                                                              |
+| BlockID    | [BlockID](#blockid)              | The blockID of the corresponding block.                              | Must adhere to the validation rules of [BlockID](#blockid).                                              |
+| Signatures | Array of [CommitSig](#commitsig) | Array of commit signatures that correspond to current validator set. | Length of signatures must be > 0 and adhere to the validation of each individual [Commitsig](#commitsig) |
 
 ## CommitSig
 
 `CommitSig` represents a signature of a validator, who has voted either for nil,
 a particular `BlockID` or was absent. It's a part of the `Commit` and can be used
 to reconstruct the vote set given the validator set.
 
-```go
-type BlockIDFlag byte
+| Name             | Type                        | Description                                                                                                  | Validation                                                        |
+|------------------|-----------------------------|---------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------|
+| BlockIDFlag      | [BlockIDFlag](#blockidflag) | Represents the validators participation in consensus: Either voted for the block that received the majority, voted for another block, voted nil or did not vote | Must be one of the fields in the [BlockIDFlag](#blockidflag) enum |
+| ValidatorAddress | [Address](#address)         | Address of the validator                                                                                      | Must be of length 20                                              |
+| Timestamp        | [Time](#time)               | This field will vary from `CommitSig` to `CommitSig`. It represents the timestamp of the validator.               | [Time](#time)                                                     |
+| Signature        | [Signature](#signature)     | Signature corresponding to the validators participation in consensus.                                         | The length of the signature must be > 0 and < than  64            |
```

Comment:
This point was never spelled out in the old spec. We should make it more clear. Somehow, 
- a signature, 
- a vote
- height, round, blockID

belong together and there is a check to ascertain that. Somehow, I must be able to check that a signature belongs to a certain height, round, blockID. I never saw that discussed anywhere

Reply:
This addition would be better suited alongside the block. The validation of commitsig just checks the length of the signature.

Reply:
added in a secondary validation section under the execution section of the block.